modules/objdetect/test/test_qrcode_encode.cpp

```diff
@@ -381,8 +381,16 @@ TEST(Objdetect_QRCode_Encode_Decode_Structured_Append, DISABLED_regression)
         std::string symbol_set = config["symbols_set"];
 
         std::string input_info = symbol_set;
+#if defined CV_CXX11
+        // std::random_shuffle is deprecated since C++11 and removed in C++17.
+        // Use manually constructed RNG and std::shuffle instead.
+        std::random_device rand_dev;
+        // Use Mersenne Twister RNG.
+        std::mt19937 rand_gen {rand_dev()};
```

Comment:
OpenCV tests should be reproducible (otherwise we would have random failures which no-one could debug).
So we should not use "true" random input.
We should use pseudo-random generators with reproducible results.

Initial code was incorrect too.
Prefer to reuse `cv::randShuffle()` over CV_8U data.

Reply:
Yes, fixed seed should work.

Reply:
Done. Also manually seeded and tested legacy call to `std::random_shuffle` (although I don't know if it is the most suitable way to do so).